index.js

```diff
@@ -77,22 +77,38 @@ hotp.gen = function(key, opt) {
  *
  */
 hotp.verify = function(token, key, opt) {
-	var window = opt.window || 50;
-	var counter = opt.counter || 0;
-
-	// Now loop through from C to C + W to determine if there is
-	// a correct code
-	for(var i = counter - window; i <=  counter + window; ++i) {
-		opt.counter = i;
-		if(this.gen(key, opt) === token) {
-			// We have found a matching code, trigger callback
-			// and pass offset
-			return { delta: i - counter };
+	var originCounter = opt.counter;
+	try {
```

Comment:
The try-finally is being used just so opt.counter will get restored before any of the return statements within the try. Though I think it's a bit unnecessary: if you wanted the code to avoid mutating the parameter, then the function should just create its own opt object to use and mutate internally.